### DISCUSSION TITLE
dl_LOCATION KOKKOS

### DISCUSSION THREAD

**Discussion by vreville** [idefix-code/idefix#297] [Q&A] [answered]
Hi, 

I have problem compiling Idefix on a local GPU cluster (NVIDIA A100) due apparently to the dl library. It is working fine on CPUs, what should go into the dl_LOCATION flag on cmake, should it depend on the architecture ? 

Thanks, 
Victor

**Comment by glesur** [comment]
Can you try to disable libdl first? e.g. `-DKokkos_ENABLE_LIBDL=NO` when you call cmake

**Comment by vreville** [comment]
Ok I tried with and without the `-DKokkos_ENABLE_LIBDL=NO`, here is the error message : 

> nvlink warning : Skipping incompatible '/usr/lib/x86_64-linux-gnu/libdl.a' when searching for -ldl
> /usr/bin/ld: CMakeFiles/idefix.dir/setup.cpp.o: undefined reference to symbol 'exp@@GLIBC_2.29'
> /usr/bin/ld: /lib/x86_64-linux-gnu/libm.so.6: error adding symbols: DSO missing from command line
> collect2: error: ld returned 1 exit status
> make[2]: *** [CMakeFiles/idefix.dir/build.make:679: idefix] Error 1
> make[1]: *** [CMakeFiles/Makefile2:607: CMakeFiles/idefix.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2

Disabling the libdl only removes the nvcc warning. 

Victor

**Reply by glesur** [reply, marked as answer]
This is not really related to idefix, it's more an issue with your cluster that can't compile cuda code. Try to clean everything in your build directory, but I'd recommend asking to your sysadmin...

**Reply by vreville** [reply]
Yes, that's what I was afraid of...
